### DISCUSSION TITLE
[Feature] Search should be in date order and have timeline

### DISCUSSION THREAD

**Discussion by Harris-KA** [immich-app/immich#5728] [Feature Request]
### The feature

The search right now is not in date order and does not have dates taken like the timeline.
Top immich, Bottom Gphotos
<img width="600" alt="image" src="https://github.com/immich-app/immich/assets/98909780/08d5ac04-71a4-4521-94ad-bd6e558fefed">
<img width="600" alt="image" src="https://github.com/immich-app/immich/assets/98909780/db1ff4a5-9401-43cd-9941-e6d05f88692e">

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
The search results are ordered by relevance. More advanced search including sorting will be added soon.

**Reply by Harris-KA** [reply]
Ok thanks

**Reply by Tronnic** [reply]
Any ETA on this? I'm having a hard time searching for specific pictures since the order is so chaotic.

**Reply by wtahler** [reply]
Very interested in this feature as well... thanks for everything that has been done for the app! This is the biggest feature request I get from my "users" (my family).

**Comment by Trakkasure** [comment]
By relevance is great. Sometimes there are multiple sort options: relevance, prominence, date, size, type
Prominence would be if the smart search term is the focal point of the search. The more tags and the less the confidence, the less prominence

**Reply by bo0tzz** [reply]
Can you elaborate on how exactly prominence would work? I'm not sure if it applies to our smart search approach.

**Reply by Trakkasure** [reply]
It would work as a sorting mechanism from the results of the search. I'm not sure if all metadata is saved (confidence % on each tag). If so, then some math can be used to calculate better relevance. I use prominence as a term to differentiate.
If a photo has many tags from smart search, then prominence would be less. Like searching for "cat" and there is a cat, people, chairs, TV, and a dog, it would be less than if there was one cat face, or an image of many cats. Things like "grass" or other scenery images could have a lower prominence so searches have a more predictable order. For more configuration, tags can be assigned and override order of importance.

**Reply by bo0tzz** [reply]
Yeah, so Immich doesn't use explicit tags, but rather the similarity of embedding vectors.

**Reply by standuptall** [reply]
Hello. I suffer this same problem. OK, it would be nice by relevance, but if I put a description in a photo, i.e. "foobar" and then search by "foobar" through my photos, I will see plenty of results except the one I was targeting

**Reply by blublub** [reply]
I also aksed myself how search is "sorted" - If I type in beack I do get images of beaches, so the smart tagging works however I would expect some sorting and the option to choose a different sorting - i.e. sort by date whiich in general searches is a really common sorting behaviour.

**Comment by mr-rulez** [comment]
Upmost missing Immich feature - from my point of view: It should be possible to see search results in timeline mode, e.g. by switching between "sort by relevance" and "sort by time".

**Comment by msegec** [comment]
This is crazy it has been considered not important to be at least able to sort newest to oldest, like what the actual?

**Comment by msegec** [comment]
This is crazy it has been considered not important to be at least able to sort newest to oldest, like what the actual?

**Comment by Tronnic** [comment]
The absence of a time based sorting in search is one of the reasons my family won't fully switch to immich. If one is to search for a photo, the only thing most people will remember is the rough estimate when the photo was taken. Using immich search is like searching trough my 50k photos using the windows search bar. I myself can't wait for this being added.

**Comment by jf-tellier** [comment]
I don't understand why this feature is not implemented yet. It is impossible to find any invoice or relevant documents.

**Reply by standuptall** [reply]
Just because of this unimplemented feature I switched again to GPhoto, and I was not happy to do it. What is the utility if Immich is unable to find what I am searching at a given moment?

**Comment by bfeist** [comment]
Crazy that this is still a conversation. Is there an Immich fork that works on feature requests?

**Reply by bo0tzz** [reply]
What a strange comment to make, have you kept up with recent releases at all?

I suggest you go read the currently active thread for this feature request; making the context search sortable is much less straightforward than you might think.

**Reply by atlas-shrugged08** [reply]
@bfeist https://github.com/immich-app/immich/discussions/8377#discussioncomment-16464734

**Comment by mws-weekend-projects** [comment]
I implemented this feature as a general UI addition "sort latest/oldest" in this PR https://github.com/immich-app/immich/pull/31572. The family is really happy with the now really usable search results.
